common/common-blackbox-setup/src/lib.rs

```diff
@@ -270,4 +265,46 @@ impl BaseSetup {
             },
         )
     }
+
+    pub fn handle_endpoint_response_and_logs(
```

Comment:
I would split this function to handle 1) error 2) logs. You could check logs even if the tx is failed (for example execute to check confirmation+deposit back).